src/core/service.c

```diff
@@ -2578,16 +2580,18 @@ static void service_notify_cgroup_empty_event(Unit *u) {
 
         case SERVICE_START:
         case SERVICE_START_POST:
-                /* If we were hoping for the daemon to write its PID file,
-                 * we can give up now. */
-                if (s->pid_file_pathspec) {
+                if (s->type == SERVICE_NOTIFY)
+                        /* No chance of getting a ready notification anymore */
+                        service_enter_signal(s, SERVICE_FINAL_SIGTERM, SERVICE_FAILURE_PROTOCOL);
```

Comment:
Thanks for the quick reply.

I'm not completely sure that moving to another cgroup is desirable, because it makes some resource control settings totally useless. On the other hand, it's possible and, maybe, we should handle it somehow. Let's see what @poettering thinks.